enterprise/mqtt/src/main/java/io/crate/mqtt/operations/MqttIngestService.java

```diff
@@ -19,47 +19,102 @@
 package io.crate.mqtt.operations;
 
 import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableMap;
 import io.crate.action.sql.BaseResultReceiver;
 import io.crate.action.sql.Option;
 import io.crate.action.sql.ResultReceiver;
 import io.crate.action.sql.SQLOperations;
-import io.crate.action.sql.SQLOperations.Session;
+import io.crate.analyze.expressions.ExpressionAnalysisContext;
+import io.crate.analyze.expressions.ExpressionAnalyzer;
+import io.crate.analyze.relations.FieldProvider;
+import io.crate.analyze.symbol.InputColumn;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.ingestion.IngestionImplementation;
+import io.crate.ingestion.IngestionService;
+import io.crate.metadata.Functions;
+import io.crate.metadata.TableIdent;
+import io.crate.metadata.rule.ingest.IngestRule;
+import io.crate.metadata.table.Operation;
+import io.crate.operation.InputFactory;
+import io.crate.operation.RowFilter;
+import io.crate.operation.user.User;
+import io.crate.operation.user.UserManager;
+import io.crate.sql.parser.SqlParser;
+import io.crate.sql.tree.QualifiedName;
 import io.crate.types.DataType;
 import io.crate.types.DataTypes;
 import io.netty.buffer.ByteBuf;
 import io.netty.handler.codec.mqtt.MqttPublishMessage;
 import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.common.logging.Loggers;
 import org.elasticsearch.common.xcontent.NamedXContentRegistry;
 import org.elasticsearch.common.xcontent.json.JsonXContent;
 
 import javax.annotation.Nullable;
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.BiConsumer;
+import java.util.function.Predicate;
 
+public class MqttIngestService implements IngestionImplementation {
```

Comment:
Imho this could use some docs - e.g. to explain the payload mapping that happens